plugins/gestures.koplugin/main.lua

```diff
@@ -0,0 +1,1036 @@
+        ges_type = "two_finger_swipe"
+        zone = zone_fullscreen
+        direction = {northwest = true}
+    elseif ges == "two_finger_swipe_northeast" then
+        ges_type = "two_finger_swipe"
+        zone = zone_fullscreen
+        direction = {northeast = true}
+    elseif ges == "two_finger_swipe_southwest" then
+        ges_type = "two_finger_swipe"
+        zone = zone_fullscreen
+        direction = {southwest = true}
+    elseif ges == "two_finger_swipe_southeast" then
+        ges_type = "two_finger_swipe"
+        zone = zone_fullscreen
+        direction = {southeast = true}
+    elseif ges == "short_diagonal_swipe" then
+        ges_type = "swipe"
+        zone = {
+            ratio_x = 0.0, ratio_y = 0,
+            ratio_w = 1, ratio_h = 1,
+        }
+        direction = {northeast = true, northwest = true, southeast = true, southwest = true}
+        distance = "short"
+        if self.is_docless then
+            overrides = {
+                "filemanager_tap",
+                "filemanager_swipe"
+            }
+        else
+            overrides = {
+                "rolling_swipe",
+                "paging_swipe",
+            }
+        end
+    elseif ges == "spread_gesture" then
+        ges_type = "spread"
+        zone = zone_fullscreen
+    elseif ges == "pinch_gesture" then
+        ges_type = "pinch"
+        zone = zone_fullscreen
+    else return
+    end
+    self:registerGesture(ges, ges_type, zone, overrides, direction, distance)
+    -- make dummy zone to disable panning and panning_release when gesture is swipe
+    if ges_type == "swipe" and ges ~= "short_diagonal_swipe" then
+        local pan_gesture = ges.."_pan"
+        local pan_release_gesture = ges.."_pan_release"
+        self:registerGesture(pan_gesture, "pan", zone, overrides_swipe_pan, direction, distance)
+        self:registerGesture(pan_release_gesture, "pan_release", zone, overrides_swipe_pan_release, direction, distance)
+    end
+end
+
+function Gestures:registerGesture(ges, ges_type, zone, overrides, direction, distance)
+    self.ui:registerTouchZones({
+        {
+            id = ges,
+            ges = ges_type,
+            screen_zone = zone,
+            handler = function(gest)
+                if distance == "short" and gest.distance > Screen:scaleBySize(300) then return end
+                if direction and not direction[gest.direction] then return end
+
+                if ges == "multiswipe" then
+                    return self:multiswipeAction(gest.multiswipe_directions, gest)
+                end
+
+                return self:gestureAction(ges, gest)
+            end,
+            overrides = overrides,
+        },
+    })
+end
+
+function Gestures:gestureAction(action, ges)
+    if G_reader_settings:isTrue("gestures_migrated") then
+        UIManager:show(InfoMessage:new{
+            text = _("Gestures have been upgraded. You may now set more than one action per gesture."),
```

Comment:
This is a popup on the first gesture after migration to let users know about the change.

Any preference for the text?

Reply:
Short and to the point, no complaint here ;).